src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Tag.cs

```diff
@@ -9,48 +9,57 @@ namespace System.Formats.Cbor
 {
     public partial class CborReader
     {
+        /// <summary>
+        ///   Reads the next data item as a semantic tag (major type 6)
+        /// </summary>
+        /// <returns>The decoded <see cref="CborTag"/> value.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   invalid CBOR encoding data --OR--
+        ///   unexpected end of CBOR encoding data --OR--
+        ///   CBOR encoding not accepted under the current conformance level
+        /// </exception>
         public CborTag ReadTag()
         {
-            CborTag tag = PeekTagCore(out int additionalBytes);
+            CborTag tag = PeekTagCore(out int bytesRead);
 
-            if (_isConformanceLevelCheckEnabled && !CborConformanceLevelHelpers.AllowsTags(ConformanceLevel))
-            {
-                throw new FormatException("Tagged items are not permitted under the current conformance level.");
-            }
-
-            AdvanceBuffer(1 + additionalBytes);
+            AdvanceBuffer(bytesRead);
             _isTagContext = true;
             return tag;
         }
 
-        public void ReadTag(CborTag expectedTag)
-        {
-            CborTag tag = PeekTagCore(out int additionalBytes);
-
-            if (_isConformanceLevelCheckEnabled && !CborConformanceLevelHelpers.AllowsTags(ConformanceLevel))
-            {
-                throw new FormatException("Tagged items are not permitted under the current conformance level.");
-            }
-
-            if (expectedTag != tag)
-            {
-                throw new InvalidOperationException("CBOR tag mismatch.");
-            }
-
-            AdvanceBuffer(1 + additionalBytes);
-            _isTagContext = true;
-        }
-
+        /// <summary>
+        ///   Reads the next data item as a semantic tag (major type 6),
```

Comment:
PeekTag requires major type 6? :)

Reply:
Oh, yeah, because "tag" is a value concept, not the same as the major type.  Never mind.